modules/icgc-argo-workflows/payload/alignment/resources/usr/bin/main.py

```diff
@@ -0,0 +1,220 @@
+        for item in ['info', 'sampleId', 'specimenId', 'donorId', 'studyId']:
+            sample.pop(item, None)
+            sample['specimen'].pop(item, None)
+            sample['donor'].pop(item, None)
+
+    return samples
+
+
+def main(args):
+    with open(args.seq_experiment_analysis, 'r') as f:
+        seq_experiment_analysis_dict = json.load(f)
+
+    pipeline_info = {}
+    if args.pipeline_yml:
+        with open(args.pipeline_yml, 'r') as f:
+            pipeline_info = yaml.safe_load(f)
+
+    updated_pipeline_info = {}
+    for key, value in pipeline_info.items():
+        new_key = key.split(":")[-1]
+        updated_pipeline_info[new_key] = value
+
+    for key, value in updated_pipeline_info.items():
+        for sub_key, sub_value in value.items():
+            value[sub_key] = str(sub_value)
+        updated_pipeline_info[key] = value
+
+    payload = {
+        'analysisType': {
+            'name': 'sequencing_alignment'
+        },
+        'studyId': seq_experiment_analysis_dict.get('studyId'),
+        'info': {},
+        'workflow': {
+            'workflow_name': args.wf_name,
+            'workflow_version': args.wf_version,
+            'genome_build': args.genome_build,
+            'run_id': args.wf_run,
+            'session_id': args.wf_session,
+            'pipeline_info': updated_pipeline_info,
+            'inputs': [
+                {
+                    'analysis_type': 'sequencing_experiment',
+                    'input_analysis_id': seq_experiment_analysis_dict.get('analysisId')
+                }
+            ]
+        },
+        'files': [],
+        'samples': get_sample_info(seq_experiment_analysis_dict.get('samples')),
+        'experiment': {},
+        'read_group_count': seq_experiment_analysis_dict.get('read_group_count'),
+        'read_groups': seq_experiment_analysis_dict.get('read_groups')
+    }
+    if args.genome_annotation:
+        payload['workflow']['genome_annotation'] = args.genome_annotation
+
+    # pass `info` dict from seq_experiment payload to new payload
+    if 'info' in seq_experiment_analysis_dict and isinstance(seq_experiment_analysis_dict['info'], dict):
+        payload['info'] = seq_experiment_analysis_dict['info']
+    else:
+        payload.pop('info')
+
+    payload['experiment'].update(seq_experiment_analysis_dict.get('experiment', {}))
+
+    if 'library_strategy' in payload['experiment']:
+        experimental_strategy = payload['experiment'].pop('library_strategy')
+        payload['experiment']['experimental_strategy'] = experimental_strategy
+
+    # get inputs from read_group_ubam_analysis
+    for ubam_analysis in args.read_group_ubam_analysis:
+        with open(ubam_analysis, 'r') as f:
+            ubam_analysis_dict = json.load(f)
+
+        payload['workflow']['inputs'].append(
+            {
+                'analysis_type': 'read_group_ubam',
+                'input_analysis_id': ubam_analysis_dict.get('analysisId')
+            }
+        )
+
```

Comment:
This piece of code to tackle ubam_analysis can be removed.

Reply:
removed